source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -123,17 +128,61 @@ bool LocalRateLimiterImpl::requestAllowedHelper(const TokenState& tokens) const
   return true;
 }
 
-bool LocalRateLimiterImpl::requestAllowed(
+StatusOr<std::reference_wrapper<const LocalRateLimiterImpl::LocalDescriptorImpl>>
+LocalRateLimiterImpl::descriptorHelper(
     absl::Span<const RateLimit::LocalDescriptor> request_descriptors) const {
   if (!descriptors_.empty() && !request_descriptors.empty()) {
+    // The override rate limit descriptor is selected by the first full match from the request
+    // descriptors.
     for (const auto& request_descriptor : request_descriptors) {
       auto it = descriptors_.find(request_descriptor);
       if (it != descriptors_.end()) {
-        return requestAllowedHelper(*it->token_state_);
+        return *it;
       }
     }
   }
-  return requestAllowedHelper(tokens_);
+  return absl::Status(absl::StatusCode::kNotFound, "no descriptor found");
+}
+
+bool LocalRateLimiterImpl::requestAllowed(
+    absl::Span<const RateLimit::LocalDescriptor> request_descriptors) const {
+  auto descriptor = descriptorHelper(request_descriptors);
+
+  return descriptor.ok() ? requestAllowedHelper(*descriptor->get().token_state_)
+                         : requestAllowedHelper(tokens_);
+}
+
+uint32_t LocalRateLimiterImpl::maxTokens(
```

Comment:
Per the previous discussion on this maybe we should have these functions be called `maxTokensForApplicableDescriptor` or something that more clearly conveys what this does?

Reply:
I am not sure if we need this change because it would make the method names very long.

Reply:
We tend to favor longer names if it helps readability, but it's probably ok as is